File: Backend/CRM.API/CRM.API/Models/FollowUp.cs
namespace CRM.API.Models
{
    public class FollowUp
    {
        public int Id { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Type { get; set; } = "FollowUp";

        public string Status { get; set; } = "Open";

        public string? Description { get; set; }

        public int OwnerId { get; set; }

        public DateTime DueAt { get; set; }

        public string? Outcome { get; set; }

        public DateTime? CompletedAt { get; set; }

        public int? ContactId { get; set; }

        public int? DepartmentId { get; set; }

        public int? ProjectId { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public User Owner { get; set; } = null!;

        public Contact? Contact { get; set; }

        public Customer? Department { get; set; }

        public Project? Project { get; set; }
    }
}

File: Frontend/src/types/followUp.ts
export type FollowUpStatus = "Open" | "Completed" | "Cancelled";
export type FollowUpType = "FollowUp" | "Call" | "Email" | "Meeting" | "CheckIn";

export interface FollowUp {
  id: number;
  title: string;
  type: FollowUpType;
  status: FollowUpStatus;
  description?: string | null;
  ownerId: number;
  ownerName: string;
  dueAt: string;
  outcome?: string | null;
  completedAt?: string | null;
  contactId?: number | null;
  contactName?: string | null;
  departmentId?: number | null;
  departmentName?: string | null;
  projectId?: number | null;
  projectName?: string | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface FollowUpRequest {
  title: string;
  type: FollowUpType;
  description?: string | null;
  ownerId: number;
  dueAt: string;
  contactId?: number | null;
  departmentId?: number | null;
  projectId?: number | null;
}
